gradebookng/tool/src/java/org/sakaiproject/gradebookng/business/GradebookNgBusinessService.java

```diff
@@ -266,16 +266,45 @@ private Gradebook getGradebook(String siteId) {
 	}
 	
 	/**
-	 * Get a list of assignments in the gradebook in the specified site that the current user is allowed to access, sorted by sort order
-	 * 
+	 * Get a list of assignments in the gradebook in the specified site that the current user is allowed to access, sorted by default sort order
+	 *
 	 * @param siteId the siteId
 	 * @return a list of assignments or null if no gradebook
 	 */
 	public List<Assignment> getGradebookAssignments(String siteId) {
```

Comment:
As you suggested, it looks we use all 3 different signatures in different spots: `getGradebookAssignments()`, `getGradebookAssignments(siteId)` and I use `getGradebookAssignments(settings.isCategoriesEnabled())` which doesn't have the siteId handy but does have the settings. 

I've introduced the new signature `public List<Assignment> getGradebookAssignments(String siteId, boolean sortByCategory)` which is the main method for all. 

Happy to refactor to simplify! Just let me know what you'd like.